runtime/drivers/duckdb/connectors.go

```diff
@@ -59,13 +60,47 @@ func (c *connection) ingest(ctx context.Context, env *connectors.Env, source *co
 
 	appendToTable := false
 	summary := &drivers.IngestionSummary{}
+	var tableSchema map[string]string
 	for iterator.HasNext() {
 		files, err := iterator.NextBatch(_iteratorBatch)
 		if err != nil {
 			return nil, err
 		}
 
-		if err := c.ingestIteratorFiles(ctx, source, files, appendToTable); err != nil {
+		from, err := sourceReader(files, source.Properties)
+		if err != nil {
+			return nil, err
+		}
+
+		var query string
+		if appendToTable {
+			srcSchema, newSchema, err := c.updateSchema(ctx, from, files, source, tableSchema)
+			if err != nil {
+				return nil, fmt.Errorf("failed to update schema %w", err)
+			}
+
+			tableSchema = newSchema
+			if srcSchema != nil {
+				// add columns and their datatypes to ensure the datatypes are not inferred again
+				var ingestionProps map[string]any
+				if duckDBProps, ok := source.Properties["duckdb"].(map[string]any); ok {
+					ingestionProps = copyMap(duckDBProps)
+				} else {
+					ingestionProps = map[string]any{}
+				}
+				ingestionProps["columns"] = schemaToDuckDBColumnsProp(srcSchema)
```

Comment:
Do we want to allow column addition in this case ?

Reply:
I remember now, I didn't handle this case since if `columns` (or `types` or `dtypes`) is specified the inference would ensure that the returned schema is same as what is specified in `columns` in cases when there are more columns in new file. In cases when new file has less columns it would just error out (not allowed by duckdb).
Hence I feel its just convenient to not add additional checks and increase complexity here and rather let the queries run (which should be fast since column inference doesn't need to happen).